googletest/include/gtest/internal/gtest-port.h

```diff
@@ -920,18 +920,22 @@ using ::std::tuple_size;
 #endif  // GTEST_HAS_SEH
 
 #ifdef _MSC_VER
-
 # if GTEST_LINKED_AS_SHARED_LIBRARY
 #  define GTEST_API_ __declspec(dllimport)
 # elif GTEST_CREATE_SHARED_LIBRARY
 #  define GTEST_API_ __declspec(dllexport)
 # endif
-
-#endif  // _MSC_VER
-
-#ifndef GTEST_API_
-# define GTEST_API_
-#endif
+#elif __GNUC__ >= 4 || defined(__clang__)
+# if GTEST_CREATE_SHARED_LIBRARY
+#  define GTEST_API_ __attribute__((visibility ("default")))
+# else
+#  define GTEST_API_
+# endif
+#else
+# ifndef GTEST_API_
```

Comment:
The ifelse chain is hard to follow. How about moving this fallback out of the entire staircase, leaving it as an unconditional cleanup after the last #endif ?